src/webui/www/private/preferences_content.html

```diff
@@ -211,40 +211,26 @@
         <legend>QBT_TR(Global Rate Limits)QBT_TR[CONTEXT=OptionsDialog]</legend>
         <table>
             <tr>
-                <td>
-                    <input type="checkbox" id="up_limit_checkbox" onclick="updateUpLimitEnabled();" />
-                    <label for="up_limit_checkbox">QBT_TR(Upload:)QBT_TR[CONTEXT=OptionsDialog]</label>
-                </td>
-                <td><input type="text" id="up_limit_value" style="width: 4em;" />&nbsp;&nbsp;QBT_TR(KiB/s)QBT_TR[CONTEXT=OptionsDialog]</td>
+                <td><input type="number" id="up_limit_value" style="width: 4em;" min="0" />&nbsp;&nbsp;QBT_TR(KiB/s)QBT_TR[CONTEXT=OptionsDialog]</td>
             </tr>
             <tr>
-                <td>
-                    <input type="checkbox" id="dl_limit_checkbox" onclick="updateDlLimitEnabled();" />
-                    <label for="dl_limit_checkbox">QBT_TR(Download:)QBT_TR[CONTEXT=OptionsDialog]</label>
-                </td>
-                <td><input type="text" id="dl_limit_value" style="width: 4em;" />&nbsp;&nbsp;QBT_TR(KiB/s)QBT_TR[CONTEXT=OptionsDialog]</td>
+                <td><input type="number" id="dl_limit_value" style="width: 4em;" min="0" />&nbsp;&nbsp;QBT_TR(KiB/s)QBT_TR[CONTEXT=OptionsDialog]</td>
             </tr>
         </table>
+        <p>QBT_TR(0 means unlimited)QBT_TR[CONTEXT=OptionsDialog]</p>
     </fieldset>
 
     <fieldset class="settings">
         <legend>QBT_TR(Alternative Rate Limits)QBT_TR[CONTEXT=OptionsDialog]</legend>
         <table>
             <tr>
-                <td>
-                    <input type="checkbox" id="alt_up_limit_checkbox" onclick="updateAltUpLimitEnabled();" />
-                    <label for="alt_up_limit_checkbox">QBT_TR(Upload:)QBT_TR[CONTEXT=OptionsDialog]</label>
-                </td>
-                <td><input type="text" id="alt_up_limit_value" style="width: 4em;" />&nbsp;&nbsp;QBT_TR(KiB/s)QBT_TR[CONTEXT=OptionsDialog]</td>
+                <td><input type="number" id="alt_up_limit_value" style="width: 4em;" min="0" />&nbsp;&nbsp;QBT_TR(KiB/s)QBT_TR[CONTEXT=OptionsDialog]</td>
```

Comment:
Why did you remove the `Upload:` and `Download:` labels?

Reply:
Oops... I didn't notice... indeed, why? That looks like a mistake.

Reply:
Sorry, stupid mistake, fixed now.

Reply:
screenshots are updated.